pkg/common/cns-lib/node/nodes.go

```diff
@@ -122,11 +121,28 @@ func (nodes *Nodes) csiNodeAdd(obj interface{}) {
 	nodeName := csiNode.Name
 	nodeUUID := k8s.GetNodeIdFromCSINode(csiNode)
 	if nodeUUID == "" {
-		log.Warnf("csiNodeAdd: nodeId is either empty. CSINode object: %v",
-			csiNode)
+		log.Warnf("csiNodeAdd: nodeId is either empty. CSINode object: %v", csiNode)
 		return
 	}
-	err := nodes.cnsNodeManager.RegisterNode(ctx, nodeUUID, nodeName)
+	err := nodes.cnsNodeManager.DiscoverNode(ctx, nodeUUID)
+	if err != nil {
+		log.Errorf("failed to find node vm using node ID on CSINode object: %v", csiNode)
+		node, err := nodes.cnsNodeManager.GetK8sNode(ctx, nodeName)
+		if err != nil {
```

Comment:
Can we check for `err != nil || node == nil` instead of checking on line 136 and silently exiting?

Reply:
done. did check `if node == nil || err != nil {`